src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -334,16 +334,20 @@ impl LogicalAgg {
                 internal_table_catalog_builder.add_column(&in_fields[include_key]);
                 column_mapping.push(include_key);
             }
-            internal_table_catalog_builder.build(in_dist_key.clone(), in_append_only)
+            internal_table_catalog_builder.build_with_column_mapping(
+                in_dist_key.clone(),
+                in_append_only,
+                column_mapping,
+            )
         };
 
-        let get_value_state_table = |value_key: usize| -> TableCatalog {
+        let get_value_state_table = |value_key_out: usize| -> TableCatalog {
             let mut internal_table_catalog_builder = TableCatalogBuilder::new();
             for &idx in &self.group_key {
                 let column_idx = internal_table_catalog_builder.add_column(&in_fields[idx]);
                 internal_table_catalog_builder.add_order_column(column_idx, OrderType::Ascending);
             }
-            internal_table_catalog_builder.add_column(&out_fields[value_key]);
+            internal_table_catalog_builder.add_column(&out_fields[value_key_out]);
             internal_table_catalog_builder.build(in_dist_key.clone(), in_append_only)
```

Comment:
I was tring to do the fix for value state, but some input dist key can not always been found in table columns